tests/test_eks/test_eks.py

```diff
@@ -797,6 +847,490 @@ def test_create_nodegroup_handles_launch_template_combinations(
+
+    with pytest.raises(ClientError) as raised_exception:
+        client.create_fargate_profile(
+            clusterName=generated_test_data.cluster_name,
+            fargateProfileName=generated_test_data.existing_fargate_profile_name,
+            **dict(FargateProfileInputs.REQUIRED)
+        )
+    count_profiles_after_test = len(
+        client.list_fargate_profiles(clusterName=generated_test_data.cluster_name)[
+            ResponseAttributes.FARGATE_PROFILE_NAMES
+        ]
+    )
+
+    count_profiles_after_test.should.equal(BatchCountSize.SMALL)
+    assert_expected_exception(raised_exception, expected_exception, expected_msg)
+
+
+@mock_eks
+def test_create_fargate_profile_throws_exception_when_cluster_not_active(
+    FargateProfileBuilder, monkeypatch
+):
+    if settings.TEST_SERVER_MODE:
+        raise SkipTest("Cant patch Cluster attributes in server mode.")
+    client, generated_test_data = FargateProfileBuilder(BatchCountSize.SMALL)
+    expected_exception = InvalidRequestException
+    expected_msg = CLUSTER_NOT_READY_MSG.format(
+        clusterName=generated_test_data.cluster_name,
+    )
+
+    with mock.patch("moto.eks.models.Cluster.isActive", return_value=False):
+        with pytest.raises(ClientError) as raised_exception:
+            client.create_fargate_profile(
+                clusterName=generated_test_data.cluster_name,
+                fargateProfileName=random_string(),
+                **dict(FargateProfileInputs.REQUIRED)
+            )
+    count_fargate_profiles_after_test = len(
+        client.list_fargate_profiles(clusterName=generated_test_data.cluster_name)[
+            ResponseAttributes.FARGATE_PROFILE_NAMES
+        ]
+    )
+
+    count_fargate_profiles_after_test.should.equal(BatchCountSize.SMALL)
+    assert_expected_exception(raised_exception, expected_exception, expected_msg)
+
+
+@mock_eks
+def test_create_fargate_profile_generates_valid_profile_arn(FargateProfileBuilder):
+    _, generated_test_data = FargateProfileBuilder()
+    expected_arn_values = [
+        PARTITIONS,
+        REGION,
+        ACCOUNT_ID,
+        generated_test_data.cluster_name,
+        generated_test_data.fargate_profile_names,
+        None,
+    ]
+
+    all_arn_values_should_be_valid(
+        expected_arn_values=expected_arn_values,
+        pattern=RegExTemplates.FARGATE_PROFILE_ARN,
+        arn_under_test=generated_test_data.fargate_describe_output[
+            FargateProfileAttributes.ARN
+        ],
+    )
+
+
+@freeze_time(FROZEN_TIME)
+@mock_eks
+def test_create_fargate_profile_generates_valid_created_timestamp(
+    FargateProfileBuilder,
+):
+    _, generated_test_data = FargateProfileBuilder()
+
+    result_time = iso_8601_datetime_without_milliseconds(
+        generated_test_data.fargate_describe_output[FargateProfileAttributes.CREATED_AT]
+    )
+
+    if settings.TEST_SERVER_MODE:
+        RegExTemplates.ISO8601_FORMAT.match(result_time).should.be.true
```

Comment:
The sure-library has a match-method, that we can use AFAIK:
```
RegExTemplates.ISO8601_FORMAT.should.match(result_time)
```
Slightly easier to read.

Reply:
Neat, I didn't know that one.  Thanks!

Reply:
Looks like the server mode tests don't like that:  https://github.com/spulec/moto/pull/4185/checks?check_run_id=3364472186#step:9:4609

Reply:
Mea culpa! I honestly thought that would work